### PR TITLE
Mega CRUDify (split, to be retired)

### PR DESCRIPTION
**Pull request by nimmolo**
This PR is now obsolete, keeping here for reference and will close soon.

### Controllers CRUDified ###
**Non-CRUD actions mostly namespaced and CRUDified**
- Observations
  _Namings, Namings/Votes, Namings/Suggestions_

### Integration tests added ###
(In progress.) In addition to updating/namespacing all the controller tests, this PR adds new integration tests for changed forms' UI.

### CODE REVIEW

**Comment by pellaea** [comment]
This is a really impressive amount of work, Nimmo.

IMPORTANT!!!!  This PR radically changes a huge amount of code that is not tested in the unit tests!!!  Let's talk about that at the next meeting.  I think we need to get everyone on board with systematically testing all the fields of all the forms this PR rewrites before we should merge it.  Maybe we can divvy it up so no one developer needs to test the whole bloody website?

This is too overwhelming for me to deal with right now.  I'm barely staying on top of the most pressing trivial requests...  Sorry!

**Comment by nimmolo** [comment]
Thanks Jason - we discussed this at the meeting and agreed on the following course of action:

Nathan will start manually checking the forms this weekend, but...

I'll make a new PR writing integration tests for each field of each of the changed forms as they exist currently.
We can merge that into `master` once the tests are done and working.

Then merge `master` into this branch and check if the forms still pass those tests.

File: app/controllers/account_controller.rb [hunk 1]
```diff
@@ -28,35 +28,18 @@
...
-#  turn_admin_on::      <tt>(R . .)</tt>
-#  turn_admin_off::     <tt>(R . .)</tt>
-#  switch_users::       <tt>(R V .)</tt>
-#  add_user_to_group::  <tt>(R V .)</tt>
-#  create_alert::       <tt>(R V .)</tt>
-#  destroy_user::       <tt>(R . .)</tt>
-#  blocked_ips::        <tt>(R V .)</tt>
-#
 #  ==== Testing
 #  test_autologin::     <tt>(L V .)</tt>
```
Comment (JoeCohen) [inline app/controllers/account_controller.rb:32]: Suggestion: update documentation at lines 8-32 to reflect updated actions

File: app/controllers/account/api_keys_controller.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      flash_error(:account_api_keys_create_failed.t + e.to_s)
+    end
+
+    def remove_api_keys
+      num_destroyed = 0
+      @user.api_keys.each do |key|
+        if params["key_#{key.id}"] == "1"
+          @user.api_keys.delete(key)
+          num_destroyed += 1
+        end
```
Comment (JoeCohen) [inline app/controllers/account/api_keys_controller.rb:65]: How about this (just a personal style choice). I think it works, but I've forgotten so much about Ruby and coding in the past few months.
```suggestion
        next unless params["key_#{key.id}"] == "1"

        @user.api_keys.delete(key)
        num_destroyed += 1
```

**Comment by JoeCohen** [comment]
It would help me if this could be broken into smaller pieces. 
Question: Is there some Git or GitHub magic that would help me figure out what lines have been changed, as opposed to being moved to a different file?

File: app/controllers/account_controller.rb [hunk 3]
```diff
@@ -28,35 +28,18 @@
...
-    :login,
-    :logout_user,
-    :reverify,
-    :send_verify,
-    :signup,
-    :verify,
+    :new,
+    :create,
     :welcome
   ]
```
Comment (JoeCohen) [inline app/controllers/account_controller.rb:40]: Is this still needed/desirable? it says that login is required before all actions, then excepts every remaining action in this controller.

**Comment by nimmolo** [comment]
With these i just kept them... happy to delete. i thought maybe someone wanted that list on hand in comments.

**Pull request closed without merging**